docker/qgisserver/geomapfish_qgisserver/accesscontrol.py

```diff
@@ -36,7 +38,17 @@ def create_session_factory(url, configuration):
     engine = sqlalchemy.create_engine(url, **configuration)
     session_factory = sessionmaker()
     session_factory.configure(bind=engine)
-    return session_factory
+
+    def get_session() -> sqlalchemy.orm.session.Session:
+        while True:
+            session = session_factory()
+            try:
+                session.execute("SELECT TRUE")  # pylint: disable=no-member
+                return session
+            except psycopg2.errors.IdleInTransactionSessionTimeout:
+                session.rollback()  # pylint: disable=no-member
+
+    return get_session
```

Comment:
Note that this can be done using https://docs.sqlalchemy.org/en/13/core/pooling.html#disconnect-handling-pessimistic